extensions/skins/conversation_v1/player.html

```diff
@@ -2,107 +2,258 @@
   <link rel="stylesheet" type="text/css"
         href="/extensions/skins/conversation_v1/static/conversation.css">
 
-  <div role="main" style="margin: 0 auto; max-width: 95%; position: relative;">
-    <div class="conversation-skin-cards-container">
-      <md-card class="conversation-skin-exploration-intro-card">
-        <img ng-src="<[introCardImageUrl]>">
-        <div class="conversation-skin-exploration-header">
-          <h3 itemprop="description" role="heading" class="protractor-test-exploration-header"><[explorationTitle]></h3>
+  <!-- Off-screen preview of future card in order to pre-determine the target height for the animation. -->
+  <md-card class="conversation-skin-tutor-card" style="position: absolute; left: -30000px; top: -30000px;">
+    <div id="futurePreview">
+      <div class="conversation-skin-tutor-card-top-section">
+        <img class="conversation-skin-initial-oppia-avatar" src="/images/general/o_black_72px.png">
+        <div angular-html-bind="upcomingContentHtml">
         </div>
-      </md-card>
+      </div>
 
-      <div class="conversation-skin-iframe-container">
-        <oppia-gadget-panel panel-contents="gadgetPanelsContents.left" class="conversation-skin-left-gadget-panel"></oppia-gadget-panel>
-        <oppia-gadget-panel panel-contents="gadgetPanelsContents.right" class="conversation-skin-right-gadget-panel"></oppia-gadget-panel>
-        <div ng-repeat="state in allResponseStates track by $index"
-             ng-class="{'conversation-skin-last-log-entry': $last}">
-          <md-card class="conversation-skin-card">
-            <div class="conversation-skin-card-feedback-toggle-container" ng-if="state.stateName && !isInPreviewMode">
-              <span class="glyphicon glyphicon-comment conversation-skin-card-feedback-toggle" popover-placement="<[isWindowNarrow() ? 'left' : 'bottom']>" popover-template="popover/feedback" popover-trigger="click" tooltip="Feedback" tooltip-placement="bottom" state-name="<[state.stateName]>">
-              </span>
-            </div>
+      <div ng-if="activeCard.interactionIsInline && upcomingInlineInteractionHtml">
```

Comment:
How does whether the current interaction is inline affect whether to use the next interaction's inline HTML in the preview?